### ISSUE TITLE
Remove custom code for checking 403s* to skip unavailable streams for a connector, identify other affected connectors

### ISSUE THREAD

**Issue by evantahler** [airbytehq/airbyte#17853]
* What are these connectors?  Make a list.
  * `source-recharge`
  * .... more! (at least 2)
* Migrate custom code to new CDK methods

**Comment by erohmensing** [comment]
[WIP]

Connectors handling 403s already:
* [Stripe](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-stripe/source_stripe/streams.py#L101-L104) - skip stream
* [Zendesk-sunshine](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-zendesk-sunshine/source_zendesk_sunshine/source.py#L70) - removed unavailable stream by commenting it out. Could be being skipped unnecessarily if this stream is now readily available


Connectors potentially expecting 403s for other reasons: 
* Amazon ads - Unclear because of [these mocked tests](https://github.com/airbytehq/airbyte/pull/5023/files?diff=unified&w=0#diff-422d940bee3a02c052097d4bdd28d9a7cea06596c8d6048e5e3ba10b2563714bR193-R206) that still exist, but look to be made redundant by changes more effectively handling errors [here](1406e0a66972002f1b440afd7d6f3cddde884507c0da056d48ce8c9422e7569aR128)?

**Comment by pedroslopez** [comment]
grooming:
- do one connector to make sure it works
- if it's many connectors, rescope this and make a list on connectors that need to be updated, potentially have GL take a look

**Comment by evantahler** [comment]
Check out https://github.com/airbytehq/oncall/issues/973 - The Facebook Marketing source is particularly bad right now with access permission failures.  Perhaps this is one of the first connectors we upgrade

cc @YowanR

**Comment by erohmensing** [comment]
I created a Google Doc with an audit of custom error handling for 403s [here](https://docs.google.com/document/d/1ghkeIhlfqx7yf6fi3sv3nbpEk7BUpIrVlPqkBmrXR0U/edit). 

TL;DR:
* I found 14 sources using HttpStreams that do custom error handling for 403s 
* Most of these were on the record-reading level, some had special cases for connection checking 
* A common spot for handling was adjusting `should_retry` to `False`. As far as I know that would be per-record, so wouldn't stop the source from trying to read more records from the stream. 
* I can't tell on my own whether there are other cases like Gitlab where you might get 403s on a record level instead of stream level - hopefully this is a good place to bring up that conversation. 
* One snag we could run into is when 403s sometimes for rate limit exceeded. Most connectors seem to handle this by checking the error code, but we could incorrectly tell the user this is a permissions issue when it is a rate limiting issue. 

Moving this to icebox for now, since this is something we want the API connectors team to take on.

**Comment by evantahler** [comment]
Thank you @erohmensing and CC @YowanR

**Comment by pedroslopez** [comment]
Looks like Snapchat also got this added recently https://github.com/airbytehq/airbyte/pull/20865

**Comment by erohmensing** [comment]
Thanks @pedroslopez - adding it!

**Comment by erohmensing** [comment]
Created a follow up issue for Alpha/beta connectors since those can be updated removing custom 403s already: https://github.com/airbytehq/airbyte/issues/22426

The rest will be handled while enabling AvailabilityStrategy for the GA connectors, which have their own separate issues. All this falls under the python connectors team, so, closing this!
